### PR TITLE
takeaway-challenge

### PR DESCRIPTION
**Pull request by AntonioDaria**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,32 +35,15 @@ So that I am reassured that my order will be delivered on time
...
-  * Implement the ability to place orders via text message.
-
-* A free account on Twilio will only allow you to send texts to "verified" numbers. Use your mobile phone number, don't worry about the customer's mobile phone.
-* Finally submit a pull request before Monday at 9am with your solution or partial solution.  However much or little amount of code you wrote please please please submit a pull request before Monday at 9am
-
+This project attempts to recreate a takeaway application to be able to place an order and receive a txt message for confirmation.
+This project is not completed and attempted until user story 3 with a minor bug during the calculation of the total price, yet to be resolved.
 
-In code review we'll be hoping to see:
+The design of the application was made using sequence diagrams and class diagrams which allowed me to create 3 classes bearing in mind the single responsibility principle.
```
Comment (soph-g) [inline README.md]: Would be nice to see the design in your README.

File: README.md [hunk 2]
```diff
@@ -48,32 +35,15 @@ So that I am reassured that my order will be delivered on time
...
 
-Reviewers will potentially be using this [code review rubric](docs/review.md).  Referring to this rubric in advance will make the challenge somewhat easier.  You should be the judge of how much challenge you want this weekend.
+The menu class would hand the calculation of the final price to a order class.
 
-Notes on Test Coverage
-------------------
+In the order class there is a bug that needs to be resolved as the price calculation it currently returns nil.
 
-You can see your [test coverage](https://github.com/makersacademy/course/blob/master/pills/test_coverage.md) when you run your tests.
+A combination of OOP techniques was used such as Dependency injection, methods Stubbing and Mocking
```
Comment (soph-g) [inline README.md]: Would like to see an example of how to interact with the code and run the tests, but the README does include good information about your approach and progress.

File: lib/menu.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+require 'order'
+class Menu
+
+  attr_reader :dishes, :order, :dishq
+
+  def initialize(dish_hash, dishes_quantity, order_class = Order)
+    @dishes = dish_hash
+    @order = order_class
+    @dishq = dishes_quantity
```
Comment (soph-g) [inline lib/menu.rb]: Naming is confusing - I thought this was a typo when I saw it being used in the code!

File: lib/menu.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
+require 'order'
+class Menu
+
+  attr_reader :dishes, :order, :dishq
+
+  def initialize(dish_hash, dishes_quantity, order_class = Order)
```
Comment (soph-g) [inline lib/menu.rb]: Good use of dependency injection

File: lib/order.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+require 'pry'
+class Order
```
Comment (soph-g) [inline lib/order.rb]: Could use private methods to make sure you are only making `add` and `total` available publicly.

File: lib/takeaway.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
+class Takeaway
```
Comment (soph-g) [inline lib/takeaway.rb]: Seems to be only printing the menu. Duplicate of functionality in Menu class, and also naming is confusing. I wasn't sure what this did until I looked at the code!

File: spec/menu_spec.rb [hunk 7]
```diff
@@ -0,0 +1,47 @@
+require 'menu'
+require 'takeaway'
+
+describe Menu do
+  subject(:menu) { described_class.new(dish_hash, dishes_quantity, order_class) }
+  let(:dishes_quantity) { { Pasta: 2, Chicken: 2 } }
+  let(:order_class) { double :order_class, total: 19.00 }
```
Comment (soph-g) [inline spec/menu_spec.rb]: Good setup and use of doubles.

File: spec/menu_spec.rb [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+      formatted_menu = "Pasta 4.50, Chicken 5.00, Veggy_wrap 3.85, Noodles 4.50"
+      expect(menu.printing_menu).to eq(formatted_menu)
+    end
+  end
+
+ context 'when the order has been placed' do
+   describe '#place_order' do
+     it 'can order a number of several dishes from the menu' do
+       allow(order_class).to receive(:add)
+       expect(menu.place_order(dishes_quantity))
```
Comment (soph-g) [inline spec/menu_spec.rb]: This expectation doesn't check for anything. Suggests that you didn't see 'red' by running the test before writing the code.

File: spec/menu_spec.rb [hunk 9]
```diff
@@ -0,0 +1,47 @@
+require 'menu'
+require 'takeaway'
+
+describe Menu do
```
Comment (soph-g) [inline spec/menu_spec.rb]: Unit tests should be grouped by method name to make them easier to read. Otherwise, testing is good. Could include edge cases to go further.

File: spec/order_spec.rb [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+     Chicken: 2
+    }
+  end
+
+  before do
+    allow(menu_class).to receive(:price).with(:Pasta).and_return(true)
+    allow(menu_class).to receive(:price).with(:Chicken).and_return(true)
+  end
+
+  describe '#add' do
```
Comment (soph-g) [inline spec/order_spec.rb]: Good grouping of tests by method
